alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportDeleteWorkflowAction.kt

```diff
@@ -150,26 +154,38 @@ class TransportDeleteWorkflowAction @Inject constructor(
                         }
                     }
 
-                    val deleteResponse = deleteWorkflow(workflow)
+                    val deleteResponse = deleteWorkflow(deleteRequest)
+                    var deleteWorkflowResponse = DeleteWorkflowResponse(deleteResponse.id, deleteResponse.version)
+
+                    val workflowMetadataId = WorkflowMetadata.getId(workflow.id)
+
+                    val metadataIdsToDelete = mutableListOf(workflowMetadataId)
+
                     if (deleteDelegateMonitors == true) {
-                        if (user != null && filterByEnabled) {
-                            // Un-stash the context
-                            withClosableContext(
-                                InjectorContextElement(
-                                    user.name.plus(UUID.randomUUID().toString()),
-                                    settings,
-                                    client.threadPool().threadContext,
-                                    user.roles,
-                                    user
-                                )
-                            ) {
-                                deleteMonitors(delegateMonitorIds, RefreshPolicy.IMMEDIATE)
-                            }
-                        } else {
-                            deleteMonitors(delegateMonitorIds, RefreshPolicy.IMMEDIATE)
-                        }
+                        val failedMonitorIds = tryDeletingMonitors(delegateMonitorIds, RefreshPolicy.IMMEDIATE)
+                        // Update delete workflow response
+                        deleteWorkflowResponse.nonDeletedMonitors = failedMonitorIds
+                        // Delete monitors workflow metadata
+                        // Monitor metadata will be in monitorId-workflowId-metadata format
+                        metadataIdsToDelete.addAll(deletableMonitors.map { MonitorMetadata.getId(it, workflowMetadataId) })
+                    }
+                    // Delete the monitors and it's metadata
+                    val deleteMonitorWorkflowMetadataResponse: BulkByScrollResponse = client.suspendUntil {
+                        DeleteByQueryRequestBuilder(client, DeleteByQueryAction.INSTANCE)
```

Comment:
Here should say:
 DeleteByQueryRequestBuilder(this, DeleteByQueryAction.INSTANCE)